rascal-lsp/src/main/rascal/lsp/lang/rascal/lsp/Templates.rsc

```diff
@@ -0,0 +1,86 @@
+@license{
+Copyright (c) 2018-2025, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+module lang::rascal::lsp::Templates
+
+import IO;
+import Location;
+import ParseTree;
+import String;
+import util::PathConfig;
+import util::Reflective;
+
+import analysis::diff::edits::TextEdits;
+
+import lang::rascal::\syntax::Rascal;
+
+LanguageFileConfig rascalConfig = fileConfig();
+
+list[FileSystemChange] newModuleTemplates(list[loc] newFiles, PathConfig(loc) getPathConfig) {
+    list[FileSystemChange] edits = [];
+    for (f <- newFiles) {
+        try {
+            // check if file is empty
+            if (!(f.extension == "rsc" && exists(f) && isBlank(f))) {
```

Comment:
how about we check if there is a `module <...>`  in there? as a user might already be typing?

Reply:
What do you mean? Aren't we out of luck when there are any contents anyway? Trying to race with a typing user seems tricky.